docs/proposals/workflow.md

```diff
@@ -0,0 +1,281 @@
+
+    // Standard object's metadata.
+    // More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#metadata.
+    api.ObjectMeta `json:"metadata,omitempty"`
+
+    // Spec defines the expected behavior of a Workflow. More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#spec-and-status.
+    Spec WorkflowSpec `json:"spec,omitempty"`
+
+    // Status represents the current status of the Workflow. More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#spec-and-status.
+    Status WorkflowStatus `json:"status,omitempty"`
+}
+
+// WorkflowList implements list of Workflow.
+type WorkflowList struct {
+    unversioned.TypeMeta `json:",inline"`
+
+    // Standard list metadata
+    // More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#metadata
+    unversioned.ListMeta `json:"metadata,omitempty"`
+
+    // Items is the list of Workflow
+    Items []Workflow `json:"items"`
+}
+```
+
+
+#### `WorkflowSpec`
+
+```go
+// WorkflowSpec contains Workflow specification
+type WorkflowSpec struct {
+    // Optional duration in seconds relative to the startTime that the job may be active
+    // before the system tries to terminate it; value must be positive integer
+    ActiveDeadlineSeconds *int64 `json:"activeDeadlineSeconds,omitempty"`
+
+    // Steps contains the vertices of the workflow graph. The key of the map is a string
+    // to uniquely identify the step. Steps order is defined by their dependencies.
+    Steps map[string]WorkflowStep `json:"steps,omitempty"`
+}
+```
+
+* `spec.steps`: is a map of `WorkflowStep`s. _Key_ of the map is a string which identifies the step.
+
+
+### `WorkflowStep`
+
+The `WorkflowStep` resource acts as a [union](https://en.wikipedia.org/wiki/Tagged_union) of `JobSpec` and `ObjectReference`.
+
+```go
+// WorkflowStep contains necessary information to identifiy the node of the workflow graph
+type WorkflowStep struct {
+    // JobTemplate contains the job specificaton that should be run in this Workflow.
+    // Only one between externalRef and jobTemplate can be set.
+    JobTemplate JobSpec `json:"jobTemplate,omitempty"`
+
+    // External contains a reference to another schedulable resource.
+    // Only one between ExternalRef and JobTemplate can be set.
+    ExternalRef api.ObjectReference `json:"externalRef,omitempty"`
+
+    // Dependecies represent dependecies of the current workflow step
+    Dependencies ObjectDependencies `json:"dependencies,omitempty"`
+}
+```
+
+* `workflowStep.jobSpec` contains the specification of the job to be executed.
+* `workflowStep.externalRef` contains a reference to external resources (for example another `Workflow`).
+*
+
+```go
+type ObjectDependencies struct {
+    // DependeciesRef is a slice of unique identifier of the step (key of the spec.steps map)
+    DependencyIDs []string `json:"dependencyIDs,omitempty"`
+    ControllerRef *ObjectReference `json:"controllerRef,omitempty"`
+    //...
```

Comment:
sorry for this and at the same time thanks. I'm going to remove this since it's not clear.